docs/adding_packages/build_and_package.md

```diff
@@ -13,24 +13,59 @@ Both methods often use build helpers to build binaries and install them into the
+
+* These compiler and linker conf from host profile should be honored and propagated to underlying build system during the build:
+  * `tools.build:cflags`
+  * `tools.build:cxxflags`
+  * `tools.build:defines`
+  * `tools.build:sharedlinklags`
+  * `tools.build:exelinkflags`
+  * `tools.apple:enable_bitcode` (only if host OS is `iOS`/`watchOS`/`tvOS`)
+
+* If host OS is Apple OS family (`macOS`/`iOS`/`watchOS`/`tvOS`), `-headerpad_max_install_names` flag should be passed to linker, unless host OS is `iOS`/`watchOS`/`tvOS` and `tools.apple:enable_bitcode` is enabled.
+
+* Multithread build (if supported by underlying build system):
+  * if some steps are sensitive to race conditions, monothread should be enforced.
+  * otherwise multithreaded build should be enabled with a number of cores controlled by `tools.build:jobs` conf from host profile if it is set, otherwise to all cores of build machine.
 
 ## Package
 
-* CMake config files must be removed (they will be generated for consumers by `cmake_find_package`, `cmake_find_package_multi`, or `CMakeDeps` generators). Use `rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))` or `rm(self, "*.cmake", os.path.join(self.package_folder, "lib"))`.
+* CMake config files must be removed. They will be generated for consumers by `CMakeDeps` generator (or legacy `cmake_find_package`/`cmake_find_package_multi` generators).
 
-* pkg-config files must be removed (they will be generated for consumers by `pkg_config` or `PkgConfigDeps` generators). Use `rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))` or `rm(self, "*.pc", os.path.join(self.package_folder, "lib"))`.
+* pkg-config files must be removed. They will be generated for consumers by `PkgConfigDeps` generator (or legacy `pkg_config` generator).
 
-* On *nix systems, executables and shared libraries should have empty `RPATH`/`RUNPATH`/`LC_RPATH`.
+* On *nix systems, executables and shared libraries should have empty `RPATH`/`RUNPATH`/`LC_RPATH`. Though, a relative path pointing inside package itself is allowed.
 
-* On macOS, install name in `LC_ID_DYLIB` section of shared libs must be `@rpath/<libfilename>`.
+* On Apple OS family:
+  * shared libs: name field of `LC_ID_DYLIB` load command must be `@rpath/<libfilename>`.
+  * shared libs & executables: name field of each `LC_LOAD_DYLIB` load command should be `@rpath/<libdependencyfilename>` (except those refering to system libs or frameworks).
 
 * Installed files must not contain absolute paths specific to build machine. Relative paths to other packages is also forbidden since relative paths of dependencies during build may not be the same for consumers. Hardcoded relative paths pointing to a location in the package itself are allowed.
+
+* Static and shared flavors of the same library must not be packaged together.
+
+* Usual package layout should be:
```

Comment:
This layout is enforced by conan center hooks, so I just translate in a formal rule what we already enforce through code.

Reply:
you may try to modify a layout, but it doesn't guarantee the resulting build will work. many projects expect to find libraries into particular locations (e.g. `Qt` wants its plug-ins within `plugins` directory, you may repackage it to the `lib`, but it will fail in runtime).